src/interface/swarm.cpp

```diff
@@ -0,0 +1,345 @@
+//========================================================================================
+// (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#include "swarm.hpp"
+#include "bvals/cc/bvals_cc.hpp"
+#include "globals.hpp" // my_rank
+#include "mesh/mesh.hpp"
+#include <cstdlib>
+#include <memory>
+#include <utility>
+#include <vector>
+
+namespace parthenon {
+
+Swarm::Swarm(const std::string label, const Metadata &metadata, const int nmax_pool_in)
+    : label_(label), m_(metadata), nmax_pool_(nmax_pool_in),
+      mask_("mask", nmax_pool_, Metadata({Metadata::Boolean})),
+      marked_for_removal_("mfr", nmax_pool_, Metadata({Metadata::Boolean})),
+      mpiStatus(true) {
+  Add("x", Metadata({Metadata::Real}));
```

Comment:
Oh I see. Swarms and swarm values are different because the swarm values are not all real. And this also adds many of the underlying needed values automatically.

Reply:
Yes that's right. I think it's fair to say that every particle needs an `x`, `y`, `z` (if you don't own a point in space then what are you), so I just made those default fields.